lib/SILOptimizer/Utils/Local.cpp

```diff
@@ -1579,4 +1579,226 @@ StaticInitCloner::clone(SingleValueInstruction *InitVal) {
   return cast<SingleValueInstruction>(ValueMap[InitVal]);
 }
 
+/// Determine InitExistential from global_addr.
+/// %3 = global_addr @$P : $*SomeP
+/// %4 = init_existential_addr %3 : $*SomeP, $SomeC
+/// %5 = alloc_ref $SomeC
+/// store %5 to %4 : $*SomeC
+/// %8 = alloc_stack $SomeP
+/// copy_addr %3 to [initialization] %8 : $*SomeP
+/// %10 = apply %9(%3) : $@convention(thin) (@in_guaranteed SomeP)
+SILValue findInitExistentialFromGlobalAddr(GlobalAddrInst *GAI,
```

Comment:
It looks like this got moved from the SILCombiner to Local.cpp. Can you split these kinds of refactorings into separate PRs, with justification?

Reply:
The reason I moved this is because Existential Specializer pass needs this apart from SILCombiner. I will create a PR for this. Should I just go ahead break this large PR into two more PRs, one with just the ExistentialTransform Pass and the other PR focusing on concrete type propagation? Ideas?

Reply:
@slavapestov https://github.com/apple/swift/pull/16950

Reply:
@slavapestov and others, I see a conflict on this PR for lib/SILOptimizer/Transforms/FunctionSignatureOpts.h because of the recent move from include to lib. Since I use some of the functionalities defined in FunctionSignatureOpts.h, is it ok to move it back to include? Any objections around this?